scripts/variantstore/wdl/GvsCreateVATfromVDS.wdl

```diff
@@ -941,7 +960,107 @@ task BigQueryLoadJson {
     }
 
     output {
-        String vat_table_name = vat_table
+        String vat_table = vat_table_name
         Boolean done = true
     }
 }
+
+task DeduplicateVatInBigQuery {
+     meta {
+         # since the WDL will not see the updated data (its getting put in a gcp bucket)
+         volatile: true
+     }
+
+     input {
+        String input_vat_table_name
+        String output_vat_table_name
+        File? nirvana_schema
+
+        String project_id
+        String dataset_name
+        String cloud_sdk_docker
+     }
+
+    command <<<
+        # Prepend date, time and pwd to xtrace log entries.
+        PS4='\D{+%F %T} \w $ '
+        set -o errexit -o nounset -o pipefail -o xtrace
+
+        echo "project_id = ~{project_id}" > ~/.bigqueryrc
+
+        DATE=86400 ## 24 hours in seconds
+
+        set +e
+        bq --apilog=false show --project_id=~{project_id} ~{dataset_name}.~{output_vat_table_name} > /dev/null
+        BQ_SHOW_RC=$?
+        set -e
+
+        if [ $BQ_SHOW_RC -ne 0 ]; then
+            echo "Creating the final vat table ~{dataset_name}.~{output_vat_table_name}"
+        else
+            bq --apilog=false rm -t -f --project_id=~{project_id} ~{dataset_name}.~{output_vat_table_name}
+        fi
+        bq --apilog=false mk --project_id=~{project_id} ~{dataset_name}.~{output_vat_table_name} ~{nirvana_schema}
+        echo "Loading data into it"
+
+        # Now we query the original VAT table and recreate it, but remove any rows that appear twice.
+
+        bq --apilog=false query --nouse_legacy_sql --destination_table=~{dataset_name}.~{output_vat_table_name} --replace --project_id=~{project_id} \
+        ' SELECT * EXCEPT(row_number) FROM (
+            SELECT
+                *,
+                row_number()
+                    over (partition by vid, transcript)
+                    row_number
+            FROM
+                `~{dataset_name}.~{input_vat_table_name}`
+            )
+            where row_number = 1'
+    >>>
+
+    runtime {
+        docker: cloud_sdk_docker
+        memory: "3 GB"
+        preemptible: 3
+        cpu: "1"
+        disks: "local-disk 100 HDD"
+    }
+
+    output {
+        String vat_table = output_vat_table_name
+        Boolean done = true
+    }
+}
+
+task GetHailScripts {
```

Comment:
probably worth unifying this with the version in `GvsCreateVds.wdl` as they are very similar and moving the result to `GvsUtils.wdl`

Reply:
Thanks - I will do that. The handling of `hail_generate_sites_only_script_path` is the substantive difference